.github/workflows/build_release.yml

```diff
@@ -0,0 +1,89 @@
+name: Cedar-agent and Docker Build
+
+on:
+  pull_request:
+    branches:
+      - main
+  release:
+    types: [published]
+
+jobs:
+  # On PR: Run Cargo tests, build, and update
+  pr_checks:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+
+      - name: Run cargo test
+        run: cargo test
+
+      - name: Run cargo build
+        run: cargo build
+
+      - name: Run cargo update
+        run: cargo update
```

Comment:
add `--locked` 
the difference is that:
- `cargo update` only updates the lock file
- `cargo update --locked` enforces that the lockfile is up to date ( see attached image )
<img width="792" alt="image" src="https://github.com/user-attachments/assets/0e596acd-0aa4-49b8-b4fe-459dc6a76208">